package.json

```diff
@@ -34,6 +35,7 @@
     "rollup-plugin-commonjs": "8.2.1",
     "rollup-plugin-node-resolve": "3.0.0",
     "tap": "~7.1.2",
+    "typescript": ">=2.2.2",
```

Comment:
this is a decently old version from some time in early 2017 from what I can see, is there a reason `3.1.1` was not chosen?

Reply:
Numerous projects still use TS 2. All of the TS language features in the deepmerge typedefs were present in TS 2, so we may as well make it clear that TS 2 users can successfully use deepmerge with these typedefs. We're just setting a baseline here, saying "if you are using TS 2.2.2 or later then these typedefs should work for you."

Reply:
Yeah, I'm fine with TS2-compatible typings.  Until TypeScript adds recursive types, I don't think there's any reason to push forward.